src/lib/types.ts

```diff
@@ -54,8 +54,13 @@ export interface SagaEnvironment<State> {
    * Wait for an action to be dispatched.
    *
    * @param actionCreator - The action creator of the action to be dispatched.
+   * @param timeout
    */
   take<Payload>(actionCreator: ActionCreator<Payload>): Promise<Payload>;
+  take<Payload>(
+    actionCreator: ActionCreator<Payload>,
+    timeout: number,
+  ): Promise<Payload | 'timeout'>;
```

Comment:
Not sure what we should return here when the timeout expires.

Reply:
I was thinking maybe we should even throw in the normal case (and if the user want's a more explicit value they have to use their own helper).

The reason would be that:

```
await take(someAction, 1000);

const data = select(getSomething);
```

The above would be the pattern I assume this to be used for, and that is wholly unsafe.

In other languages we could annotate this function such that the return value is not ignored.

(And at least in the cases in OMS that I remember, not getting the desired action in time is indeed and error and we already log on `'timeout'`.)

Reply:
Yeah and you can still execute if the timeout happens:
```
try {
  const action = await take(someAction, 1000);
} catch (error) {
  // Code for the timeout
}
```

I think throwing an error will be easier to migrate and at the and also use it.

Reply:
Agreed. Thank you for this. Gonna clean up our app code then 😊